### PR TITLE
Refactor torch focal loss

### PR DESCRIPTION
**Pull request by nrweir**
Re-implementing focal loss for PyTorch in a way that can accommodate non-logit inputs.

*IMPORTANT NOTE:* This version of the focal loss doesn't have alpha balancing. This is something that could be implemented sometime in the future, I'll make an issue for it.

### CODE REVIEW

**Comment by nrweir** [comment]
I also added a test here that ensures inputting the same numbers for `outputs` and `targets` results in a loss of `0`.
